src/main/java/io/supertokens/emailpassword/EmailPassword.java

```diff
@@ -373,76 +374,86 @@ public static void updateUsersEmailOrPassword(Main main,
-                            transaction, userId);
+        Storage[] storages = StorageLayer.getStoragesForApp(main, appIdentifier);
+        for (Storage storage : storages) {
+            EmailPasswordSQLStorage epSqlStorage = (EmailPasswordSQLStorage) storage;
+            try {
+                epSqlStorage.startTransaction(transaction -> {
+                    try {
+                        UserInfo userInfo = epSqlStorage.getUserInfoUsingId_Transaction(appIdentifier, transaction, userId);
 
-                    if (userInfo == null) {
-                        throw new StorageTransactionLogicException(new UnknownUserIdException());
-                    }
+                        if (userInfo == null) {
+                            throw new StorageTransactionLogicException(new UnknownUserIdException());
+                        }
 
-                    if (email != null) {
-                        try {
-                            storage.updateUsersEmail_Transaction(tenantIdentifier.toAppIdentifier(), transaction,
-                                    userId, email);
-                        } catch (DuplicateEmailException e) {
-                            throw new StorageTransactionLogicException(e);
+                        if (email != null) {
+                            try {
+                                epSqlStorage.updateUsersEmail_Transaction(appIdentifier, transaction,
+                                        userId, email);
+                            } catch (DuplicateEmailException e) {
+                                throw new StorageTransactionLogicException(e);
+                            }
                         }
-                    }
 
-                    if (password != null) {
-                        String hashedPassword = PasswordHashing.getInstance(main)
-                                .createHashWithSalt(tenantIdentifier.toAppIdentifier(), password);
-                        storage.updateUsersPassword_Transaction(tenantIdentifier.toAppIdentifier(), transaction, userId,
-                                hashedPassword);
-                    }
+                        if (password != null) {
+                            String hashedPassword = PasswordHashing.getInstance(main)
+                                    .createHashWithSalt(appIdentifier, password);
+                            epSqlStorage.updateUsersPassword_Transaction(appIdentifier, transaction, userId, hashedPassword);
+                        }
 
-                    storage.commitTransaction(transaction);
-                    return null;
-                } catch (TenantOrAppNotFoundException e) {
-                    throw new StorageTransactionLogicException(e);
+                        epSqlStorage.commitTransaction(transaction);
+                        return null;
+                    } catch (TenantOrAppNotFoundException e) {
```

Comment:
where will this (TenantOrAppNotFoundException) exception come from in the above block?